### PR TITLE
Add option to customize requests

### LINKED ISSUE #5: Missing documentation of addTopTop option

### PR DESCRIPTION
**Pull request by giovanisleite**
- Add option to customize requests based on axios options (https://github.com/axios/axios#request-config)
- Update documentation
- Fix #5

### CODE REVIEW

File: src/useCroods.tsx [hunk 1]
```diff
@@ -74,12 +74,11 @@ const useCroods = ({
...
-      addToTop,
-      onProgress,
-      $_requestConfig = {},
-      ...rawBody
-    }: any) => {
+    ({
+      onProgress: onUploadProgress,
+      requestConfig = {},
+      ...contextOpts
+    }) => async ({ addToTop, ...rawBody }: any) => {
```
Comment (giovanisleite) [inline src/useCroods.tsx]: Why not pass addToTop too to the contextOpts?

Comment (vractal) [reply src/useCroods.tsx]: I though about it, but wandered if there was a semantic reason to be there in the first place.. But I guess it makes sense to move it

File: src/useCroods.tsx [hunk 2]
```diff
@@ -103,7 +108,7 @@ const useCroods = ({
   )
 
   const destroy = useCallback(
-    contextOpts => async (query?: object) => {
+    contextOpts => async (query: object = {}, requestConfig: object = {}) => {
```
Comment (gustavoguichard) [inline src/useCroods.tsx]: I'm not sure `object` is permissive enough in Typescript.. I had problems with it before.
I ended up using either `any` which is bad or some version of:
```js
interface MyObj {
  someAttrIKnow?: string
  [key: string]: any
}
```

File: docs/docs/the-actions.md [hunk 3]
```diff
@@ -116,8 +116,9 @@ The save action will controll the `POST` and `PUT` requests and everything relat
 
 #### Params
 
-This method is configured by calling it with a `config` param, an object with everything you could config on [`useCroods`](/docs/use-croods-api) and then called again to pass the data to the server.
+This method is configured by calling it with a `config` param, an object with everything you could config on [`useCroods`](/docs/use-croods-api), plus two parameters: `onProgress`, a function which you can pass to handle the request progress (usually for uploads), and `addToTop`, that if set to true will add the created object at the top of the state list. Then you call the method again with the data to be sent to the server.
```
Comment (gustavoguichard) [inline docs/docs/the-actions.md]: It'd be nice to have a proper sample in this context

File: src/useCroods.tsx (hunk 1 above)
Comment (gustavoguichard) [reply src/useCroods.tsx]: not sure why... this is some legacy that we brought and never used as far as I was on it.
But probably left it explicitly due to lack of documentation.

**Pull request merged**